File: src/org/lineageos/setupwizard/SetupWizardActivity.java
/*
 * SPDX-FileCopyrightText: 2016 The CyanogenMod Project
 * SPDX-FileCopyrightText: 2017-2024 The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.setupwizard;

import static android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION;

import static org.lineageos.setupwizard.SetupWizardApp.ACTION_LOAD;
import static org.lineageos.setupwizard.SetupWizardApp.EXTRA_SCRIPT_URI;
import static org.lineageos.setupwizard.SetupWizardApp.EXTRA_WIZARD_BUNDLE;
import static org.lineageos.setupwizard.SetupWizardApp.LOGV;

import android.annotation.Nullable;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import org.lineageos.setupwizard.util.SetupWizardUtils;
import org.lineageos.setupwizard.wizardmanager.WizardManager;

public class SetupWizardActivity extends AppCompatActivity {
    private static final String TAG = SetupWizardActivity.class.getSimpleName();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (LOGV) {
            Log.v(TAG, "onCreate savedInstanceState=" + savedInstanceState);
        }
        if (SetupWizardUtils.hasLeanback(this) && SetupWizardUtils.hasGMS(this)) {
            finish();
            return;
        }
        SetupWizardUtils.enableComponent(this, WizardManager.class);
        Intent intent = new Intent(ACTION_LOAD);
        Bundle wizardBundle = new Bundle();
        if (SetupWizardUtils.isOwner()) {
            wizardBundle.putString(EXTRA_SCRIPT_URI, getString(R.string.lineage_wizard_script_uri));
        } else if (SetupWizardUtils.isManagedProfile(this)) {
            wizardBundle.putString(EXTRA_SCRIPT_URI,
                    getString(R.string.lineage_wizard_script_managed_profile_uri));
        } else {
            wizardBundle.putString(EXTRA_SCRIPT_URI,
                    getString(R.string.lineage_wizard_script_user_uri));
        }
        intent.putExtra(EXTRA_WIZARD_BUNDLE, wizardBundle);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | FLAG_GRANT_READ_URI_PERMISSION);
        intent.setPackage(getPackageName());
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/org/lineageos/setupwizard/SetupWizardActivity.kt
/*
 * SPDX-FileCopyrightText: 2016 The CyanogenMod Project
 * SPDX-FileCopyrightText: The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.setupwizard

import android.content.Intent
import android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import org.lineageos.setupwizard.SetupWizardApp.Companion.ACTION_LOAD
import org.lineageos.setupwizard.SetupWizardApp.Companion.EXTRA_SCRIPT_URI
import org.lineageos.setupwizard.SetupWizardApp.Companion.EXTRA_WIZARD_BUNDLE
import org.lineageos.setupwizard.SetupWizardApp.Companion.LOGV
import org.lineageos.setupwizard.util.SetupWizardUtils
import org.lineageos.setupwizard.wizardmanager.WizardManager

class SetupWizardActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (LOGV) {
            Log.v(TAG, "onCreate savedInstanceState=$savedInstanceState")
        }
        if (SetupWizardUtils.hasLeanback(this) && SetupWizardUtils.hasGMS(this)) {
            finish()
            return
        }
        SetupWizardUtils.enableComponent(this, WizardManager::class.java)
        val scriptUri =
            when {
                SetupWizardUtils.isOwner() -> R.string.lineage_wizard_script_uri
                SetupWizardUtils.isManagedProfile(this) ->
                    R.string.lineage_wizard_script_managed_profile_uri
                else -> R.string.lineage_wizard_script_user_uri
            }
        val wizardBundle = Bundle().apply { putString(EXTRA_SCRIPT_URI, getString(scriptUri)) }
        val intent =
            Intent(ACTION_LOAD).apply {
                putExtra(EXTRA_WIZARD_BUNDLE, wizardBundle)
                flags = Intent.FLAG_ACTIVITY_NEW_TASK or FLAG_GRANT_READ_URI_PERMISSION
                setPackage(packageName)
            }
        startActivity(intent)
        finish()
    }

    companion object {
        private val TAG: String = SetupWizardActivity::class.java.simpleName
    }
}
